webpack/assets/javascripts/react_app/components/BookmarkForm/BookmarkForm.js

```diff
@@ -21,56 +28,182 @@ const BookmarkForm = ({
   const existsNamesRegex = new RegExp(
     `^(?!(${bookmarks.map(({ name }) => name).join('|')})$).+`
   );
-  const bookmarkFormSchema = Yup.object().shape({
-    name: Yup.string()
-      .max(...maxLengthMsg(254))
-      .required(requiredMsg())
-      .matches(existsNamesRegex, {
-        excludeEmptyString: true,
-        message: __('name already exists'),
-      }),
-    query: Yup.string()
-      .max(...maxLengthMsg(4096))
-      .required(requiredMsg()),
-  });
 
-  const handleSubmit = (values, actions) =>
+  const QUERY_MAX_LENGTH = 4096;
+  const NAME_MAX_LENGTH = 254;
+  const SUCCESS_TYPE = 'success';
+  const ERROR_TYPE = 'error';
+
+  const nameTooLongMsg = sprintf(
+    __('name is too long, max %s'),
+    NAME_MAX_LENGTH
+  );
+  const nameExists = __('name already exists');
+  const queryTooLongMsg = sprintf(
+    __('Query too long, max %s'),
+    QUERY_MAX_LENGTH
+  );
+  const blankInputErrMsg = __("Can't be blank");
+
+  const [name, setName] = useState('');
+  const [query, setQuery] = useState(searchQuery);
+  const [isPublic, setIsPublic] = useState(true);
+
+  const [nameErrorMsg, setNameErrorMsg] = useState('');
+  const [nameValidated, setNameValidated] = useState('');
+  const [queryErrorMsg, setQueryErrorMsg] = useState('');
+  const [queryValidated, setQueryValidated] = useState('');
+
+  const validateName = value => {
+    if (value.length > NAME_MAX_LENGTH) {
+      setNameValidated(ERROR_TYPE);
+      setNameErrorMsg(nameTooLongMsg);
+    } else if (!existsNamesRegex.test(value)) {
+      setNameValidated(ERROR_TYPE);
+      setNameErrorMsg(nameExists);
+    } else {
+      setNameValidated(SUCCESS_TYPE);
+    }
+  };
+
+  const validateQueryLength = value => {
+    if (value.length > QUERY_MAX_LENGTH) {
+      setQueryValidated(ERROR_TYPE);
+      setQueryErrorMsg(queryTooLongMsg);
+    } else if (value.length < 1) {
```

Comment:
Similarly, there's a `minlength` or `required` field. You already pass `isRequired` so doesn't that already validate the field is non-empty?